package/MDAnalysis/analysis/psa.py

```diff
@@ -591,6 +595,30 @@ def dist_mat_to_vec(N, i, j):
         raise ValueError(err_str)
 
 
+def validate_dist_mat_to_vec_inputs(N, i, j):
+    """
+    Validate the inputs of the function dist_mat_to_vec else raise an exception
+    :param N: int
+    :param i: int
+    :param j: int
+    :return: null
+    """
+
+    # Validating if the inputs are integer
+    try:
+        NValue = int(N)
+        iValue = int(i)
+        jValue = int(j)
+    except ValueError:
+        raise ValueError
```

Comment:
Integer evaluation isn't done like that. Here we just check if the input can be converted to an integer. So `'1'` would still be a valid input here with this code. The correct test is `isinstance(var, [class])`.  So here it would be

```
if not (isinstance(N, int) or isinstance(i, int) or isinstance(j, int)):
    raise ValueError("N,i,j all must be of type int")
```

Here you should also check if this still works with numpy integer types.